### PR TITLE
feat: Add salt flag to `forc deploy`, maturity flag to `deploy` and `run`

### PR DESCRIPTION
**Pull request by mitchmindtree**
## Description

This allows for specifying a `--salt` for the top-level contract member when running `forc deploy`. Right now, we only enable this for *packages*, and not for workspaces with more than one package due to some ambiguities that arise (see comment note in code). I'll open an issue to track relaxing this constraint a little. Edit: opened https://github.com/FuelLabs/sway/issues/4117.

I noticed we also had no way of specifying maturity, so have added a `--maturity` flag for this as well.

These flags are shared with the `forc-tx` crate in order to try and provide some consistency (in help output and flag representation) across our CLI tooling.

Ideally we'd like to ship this feature in the Sway release that ends up in the fuelup beta-3 toolchain - otherwise users following the quick-start guide will be unable to re-deploy the same contract to the network as it will already exist - they'll need to be able to provide a salt.

## Checklist

- [x] I have linked to any relevant issues.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation where relevant (API docs, the reference, and the Sway book).
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] I have added (or requested a maintainer to add) the necessary `Breaking*` or `New Feature` labels where relevant.
- [x] I have done my best to ensure that my PR adheres to [the Fuel Labs Code Review Standards](https://github.com/FuelLabs/rfcs/blob/master/text/code-standards/external-contributors.md).
- [x] I have requested a review from the relevant team or maintainers.

### CODE REVIEW

**Review by otrho — APPROVED** [review]

**Review by JoshuaBatty — DISMISSED** [review]

File: forc-plugins/forc-client/src/op/deploy.rs [hunk 1]
```diff
@@ -64,7 +84,7 @@ pub async fn deploy_pkg(
     let client = FuelClient::new(node_url)?;
 
     let bytecode = compiled.bytecode.clone().into();
-    let salt = Salt::new([0; 32]);
+    let salt = command.salt.salt.unwrap_or_default();
```
Comment (Voxelot) [inline forc-plugins/forc-client/src/op/deploy.rs:87]: Would it make sense to support the ability to generate a random salt for the user? Would make things like QuickStart doc examples easier if users could just copy paste commands without needing to customize arguments etc

Comment (mitchmindtree) [reply forc-plugins/forc-client/src/op/deploy.rs:87]: Yeah that sounds useful :+1: I'll add a `--random-salt` flag now unless there's any objections?

**Comment by mitchmindtree** [comment]
Okydoke, rebased and added @Voxelot's suggestion for a `--random-salt` option!

**Review by kayagokalp — APPROVED** [review]
utACK ✅

**Review by Voxelot — APPROVED** [review]

**Pull request merged**